_posts/2019-09-30-introducing-arrow-flight.md

```diff
@@ -0,0 +1,278 @@
+* Nodes in a distributed cluster can take on different roles. For example, a
+  subset of nodes might be responsible for planning queries while other nodes
+  exclusively fulfill data stream (`DoGet` or `DoPut`) requests.
+
+Here is an example diagram of a multi-node architecture with split service
+roles:
+
+<div align="center">
+<img src="{{ site.baseurl }}/img/20191014_flight_complex.png"
+     alt="Flight Complex Architecture"
+     width="60%" class="img-responsive">
+</div>
+
+## Actions: Extending Flight with application business logic
+
+While the `GetFlightInfo` request supports sending opaque serialized commands
+when requesting a dataset, a client may need to be able to ask a server to
+perform other kinds of operations. For example, a client may request for a
+particular dataset to be "pinned" in memory so that subsequent requests from
+other clients are served faster.
+
+A Flight service can thus optionally define "actions" which are carried out by
+the `DoAction` RPC. An action request contains the name of the action being
+performed and optional serialized data containing further needed
+information. The result of an action is a gRPC stream of opaque binary results.
+
+An example action would be the command `'ListDatasets'` which could return a
+stream of dataset names that are available on that server.
+
+Note that it is not required for a server to implement any actions, and actions
+need not return results.
+
+## Encryption and Authentication
+
+Flight supports encryption out of the box using gRPC's built in TLS / OpenSSL
+capabilities.
+
+For authentication, there are extensible authentication handlers for the client
+and server that permit simple authentication schemes (like user and password)
+as well as more involved authentication such as Kerberos. The Flight protocol
+comes with a built-in `BasicAuth` so that user/password authentication can be
+implemented out of the box without custom development.
+
+## Middleware and Tracing
+
+gRPC has the concept of "interceptors" which have allowed us to develop
+developer-defined "middleware" that can provide instrumentation of or telemetry
+for incoming and outgoing requests. One such framework for such instrumentation
+is [OpenTracing][6].
+
+Note that middleware functionality is one of the newest areas of the project
+and is only currently available in the project's master branch.
+
+## gRPC, but not only gRPC
+
+We specify server locations for `DoGet` requests using RFC 3986 compliant
+URIs. For example, TLS-secured gRPC may be specified like
+`grpc+tls://$HOST:$PORT`.
+
+While we think that using gRPC for the "command" layer of Flight servers makes
+sense, we may wish to support data transport layers other than TCP such as
+[RDMA][7]. While some design and development work is required to make this
+possible, the idea is that gRPC could be used to coordinate get and put
+transfers which may be carried out on protocols other than TCP.
+
+## Getting Started and What's Next
```

Comment:
I think it would be helpful to link to some real world examples. Two that we have are:

Connector to access Dremio using Arrow Flight instead of ODBC: 
https://github.com/dremio-hub/dremio-flight-connector
20-50x performance improvements

Arrow Flight Data source for Apache Spark
https://github.com/rymurr/flight-spark-source

@rymurr, can you add a readme for the second link?

Reply:
done!